### PR TITLE
server built with falcon with can parse post message with fields

### PR DESCRIPTION
**Pull request by alextikh**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,16 @@
+all: server test clean
+
+server:
+	@echo "\nSetting up the server\n"
+	@gunicorn server &
+	@sleep 1
+
+test:
```
Comment (fuxpavel) [inline Makefile:3]: test should depend on server

File: login.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import falcon
+import json
+
+class Login(object):
+
+    def on_post(self, req, resp):
+        body = req.stream.read()
+        userdata = json.loads(body)
+	username = userdata['username']
```
Comment (fuxpavel) [inline login.py]: you have an indentation problem, please use 4 spaces always

File: login.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+	password = userdata['password']
+        resp.body = 'username: {:s}, password: {:s}'.format(username,
```
Comment (fuxpavel) [inline login.py]: the response body should be json too

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+	@echo "\n\n\nRunning test:\n"
+	@python test.py
```
Comment (fuxpavel) [inline Makefile]: you forgot to add the test.py

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+	@sleep 1
```
Comment (fuxpavel) [inline Makefile]: why do we need sleep?

Comment (alextikh) [inline Makefile]: Overwise, the whole iutput is messy: test.py begins to print, followed up by output from gunicorn. Hard to keep track of what's going on.

File: login.py (hunk 2 above)
Comment (fuxpavel) [inline login.py]: I think you still have a problem here

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,15 @@
+clean: test server
+	@echo "\n\n\nShutting down the server\n"
```
Comment (fuxpavel) [inline Makefile]: you should add all target that runs everything
I think that the kill should be part of the same target
I think you should create a tarter called run_tests or something like that...

Comment (fuxpavel) [inline Makefile]: Here is a good example:
https://github.com/Stratoscale/upseto/blob/master/Makefile

File: test.py [hunk 7]
```diff
@@ -0,0 +1,9 @@
+import requests
+
+url = 'http://localhost:8000/login'
+username = raw_input('username: ')
```
Comment (fuxpavel) [inline test.py]: there shouldn't be any code out side of a function, you should use python unittest module for testing

File: test_server.py [hunk 8]
```diff
@@ -0,0 +1,88 @@
+import requests
+import json
+import unittest
+import subprocess
+import socket
+import time
+
+
+class Server:
```
Comment (fuxpavel) [inline test_server.py:9]: please inherit from object

File: test_server.py [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+            except socket.error as e:
+                status = False
+            else:
+                status = True
+            time.sleep(delay)
+        sock.close()
+        return status
+
+
+class ServerCommunication:
```
Comment (fuxpavel) [inline test_server.py:39]: here too

**Pull request merged**